OpenRA.Game/Graphics/HardwareCursor.cs

```diff
@@ -123,16 +149,46 @@ void Update()
 			}
 		}
 
-		public void Render(Renderer renderer) { }
+		public void Render(Renderer renderer)
+		{
+			if (cursor.Name == null || isLocked == false)
+				return;
+
+			var cursorSequence = cursorProvider.GetCursorSequence(cursor.Name);
+			var cursorSprite = sprites[cursor.Name][frame];
+			var cursorSize = CursorProvider.CursorViewportZoomed ? 2.0f * cursorSprite.Size : cursorSprite.Size;
```

Comment:
`CursorViewportZoomed` is only valid for software cursors, so using this here (settings -> disable hardware cursors, restart, settings -> pixel doubling + cursor doubling, enable hardware cursors, restart) produces a fun situation where the cursor suddenly doubles in size when locked.

We also need to account for `CursorViewportZoomed` when rendering the direction arrows...